flycheck.el

```diff
@@ -2245,34 +2281,46 @@ Display a new buffer listing all syntax checkers that could be
 applicable in the current buffer.  For each syntax checkers,
 possible problems are shown."
   (interactive)
+  ;; Save to make sure checkers that only work on saved buffers will pass the
+  ;; verification
   (when (and (buffer-file-name) (buffer-modified-p))
-    ;; Save the buffer
     (save-buffer))
```

Comment:
Ugh, I didn't realize we did that.  Maybe we should do it through `save-some-buffers` instead?
I'd be happy with just a FIXME; no need to fix it now.

Reply:
Added a FIXME.  `save-some-buffers` would let the user choose, but then we would have a few `:predicate` failing due to `flycheck-buffer-saved-p`.  Ideally we would annotate these in the verify buffer, but how do we know that `flycheck-buffer-saved-p` is used in the predicate?  Macro magic in `define-checker`?

Reply:
I think we can't know; the best we could do would be to show a warning if the current buffer is unsaved.